include/net/nrf_cloud_defs.h

```diff
@@ -0,0 +1,202 @@
+#define NRF_CLOUD_LOCATION_JSON_KEY_WIFI_RSSI	"signalStrength"
+#define NRF_CLOUD_LOCATION_JSON_KEY_WIFI_SSID	"ssid"
+#define NRF_CLOUD_LOCATION_JSON_KEY_LAT		"lat"
+#define NRF_CLOUD_LOCATION_JSON_KEY_LON		"lon"
+#define NRF_CLOUD_LOCATION_JSON_KEY_UNCERT	"uncertainty"
+#define NRF_CLOUD_LOCATION_TYPE_VAL_MCELL	"MCELL"
+#define NRF_CLOUD_LOCATION_TYPE_VAL_SCELL	"SCELL"
+#define NRF_CLOUD_LOCATION_TYPE_VAL_WIFI	"WIFI"
+
+/* P-GPS */
+#define NRF_CLOUD_JSON_PGPS_PRED_COUNT		"predictionCount"
+#define NRF_CLOUD_JSON_PGPS_INT_MIN		"predictionIntervalMinutes"
+#define NRF_CLOUD_JSON_PGPS_GPS_DAY		"startGpsDay"
+#define NRF_CLOUD_JSON_PGPS_GPS_TIME		"startGpsTimeOfDaySeconds"
+#define NRF_CLOUD_PGPS_RCV_ARRAY_IDX_HOST	0
+#define NRF_CLOUD_PGPS_RCV_ARRAY_IDX_PATH	1
+#define NRF_CLOUD_PGPS_RCV_REST_HOST		"host"
+#define NRF_CLOUD_PGPS_RCV_REST_PATH		"path"
+
+/* A-GPS */
+#define NRF_CLOUD_JSON_KEY_ELEVATION_MASK	"mask"
+#define NRF_CLOUD_JSON_KEY_AGPS_TYPES		"types"
+
+/* FOTA */
+#define NRF_CLOUD_FOTA_TYPE_MODEM_DELTA		"MODEM"
+#define NRF_CLOUD_FOTA_TYPE_MODEM_FULL		"MDM_FULL"
+#define NRF_CLOUD_FOTA_TYPE_BOOT		"BOOT"
+#define NRF_CLOUD_FOTA_TYPE_APP			"APP"
+#define NRF_CLOUD_FOTA_REST_KEY_JOB_DOC		"jobDocument"
+#define NRF_CLOUD_FOTA_REST_KEY_JOB_ID		"jobId"
+#define NRF_CLOUD_FOTA_REST_KEY_PATH		"path"
+#define NRF_CLOUD_FOTA_REST_KEY_HOST		"host"
+#define NRF_CLOUD_FOTA_REST_KEY_TYPE		"firmwareType"
+#define NRF_CLOUD_FOTA_REST_KEY_SIZE		"fileSize"
+#define NRF_CLOUD_FOTA_REST_KEY_VER		"version"
+/** Current FOTA version number */
+#define NRF_CLOUD_FOTA_VER			2
+
+/* REST */
+#define NRF_CLOUD_REST_ERROR_CODE_KEY		"code"
+#define NRF_CLOUD_REST_ERROR_MSG_KEY		"message"
+
+/* GNSS - PVT */
+#define NRF_CLOUD_JSON_GNSS_PVT_KEY_LAT		"lat"
+#define NRF_CLOUD_JSON_GNSS_PVT_KEY_LON		"lng"
+#define NRF_CLOUD_JSON_GNSS_PVT_KEY_ACCURACY	"acc"
+#define NRF_CLOUD_JSON_GNSS_PVT_KEY_ALTITUDE	"alt"
+#define NRF_CLOUD_JSON_GNSS_PVT_KEY_SPEED	"spd"
+#define NRF_CLOUD_JSON_GNSS_PVT_KEY_HEADING	"hdg"
+
+/* Device Info */
+#define NRF_CLOUD_DEVICE_JSON_KEY_NET_INF	"networkInfo"
+#define NRF_CLOUD_DEVICE_JSON_KEY_SIM_INF	"simInfo"
+#define NRF_CLOUD_DEVICE_JSON_KEY_DEV_INF	"deviceInfo"
+
+/* ALERTS */
+#define NRF_CLOUD_JSON_ALERT_SEQUENCE		"seq"
+#define NRF_CLOUD_JSON_ALERT_DESCRIPTION	"desc"
+#define NRF_CLOUD_JSON_ALERT_TYPE		"type"
+#define NRF_CLOUD_JSON_ALERT_VALUE		"value"
+
+/* Settings Module */
+/** nRF Cloud's string identifier for persistent settings */
+#define NRF_CLOUD_SETTINGS_NAME			"nrf_cloud"
+#define NRF_CLOUD_SETTINGS_FOTA_KEY		"fota"
+#define NRF_CLOUD_SETTINGS_FOTA_JOB		"job"
+/** String used when defining a settings handler for FOTA */
+#define NRF_CLOUD_SETTINGS_FULL_FOTA		NRF_CLOUD_SETTINGS_NAME \
+						"/" \
+						NRF_CLOUD_SETTINGS_FOTA_KEY
+/** String used when saving FOTA job info to settings */
+#define NRF_CLOUD_SETTINGS_FULL_FOTA_JOB	NRF_CLOUD_SETTINGS_FULL_FOTA \
+						"/" \
+						NRF_CLOUD_SETTINGS_FOTA_JOB
```

Comment:
Out of scope for this PR, but why do we expose these? They feel like they should be private members of `nrf_cloud_fota` (since they define that library's reservation in the Zephyr settings subsystem) -- Do you think we should write a ticket to make these private, or am I misunderstanding something?

Reply:
https://github.com/nrfconnect/sdk-nrf/blob/860a4fe038c9edcdd53d05285f82f2aa35a49d05/samples/nrf9160/nrf_cloud_rest_fota/Kconfig#L34

https://github.com/nrfconnect/sdk-nrf/blob/860a4fe038c9edcdd53d05285f82f2aa35a49d05/samples/nrf9160/nrf_cloud_rest_fota/src/main.c#L25

Reply:
I guess I question whether we need that sample to have this customizable at all -- In fact I don't fully follow why that sample is interacting with the settings subsystem, I would have assumed the nrf_cloud_fota lib took care of that

Reply:
that is the REST sample that doesn't use nrf_cloud_fota.
if any application wants to update to something that uses nrf_cloud_fota, it needs to know where its settings are.

Reply:
Oh, duh, I understand now. My bad, I got lead astray a bit by the name, but I see now that it's a non-`nrf_cloud_fota` demonstration of nRFCloud's FOTA features